app/domain/authentication/authn_azure/authenticator.rb

```diff
@@ -6,35 +6,77 @@ module AuthnAzure
     Err = Errors::Authentication::AuthnAzure
     Log = LogMessages::Authentication::AuthnAzure
     # Possible Errors Raised:
-    # TODO: Add errors
+    # TokenFieldNotFoundOrEmpty, MissingRequestParam
 
     Authenticator = CommandClass.new(
       dependencies: {
-        fetch_authenticator_secrets: Authentication::Util::FetchAuthenticatorSecrets.new,
-        logger:                      Rails.logger
+        fetch_authenticator_secrets:   Authentication::Util::FetchAuthenticatorSecrets.new,
+        verify_and_decode_token:       Authentication::AuthnOidc::VerifyAndDecodeToken.new,
+        logger:                        Rails.logger
       },
-      inputs: [:authenticator_input]
+      inputs:       [:authenticator_input]
     ) do
 
       extend Forwardable
-      def_delegators :@authenticator_input, :service_id, :authenticator_name, :account, :username, :request, :credentials
+      def_delegators :@authenticator_input, :service_id, :authenticator_name, :account, :credentials
+
+      JWT_FIELD_NAME = "jwt"
 
       def call
+        validate_credentials_include_azure_token
+        validate_azure_token
+        validate_required_token_fields_exist
         true
       end
 
       private
 
-      #def validate_azure_token
-      #
-      #end
-      #
-      #def validate_application_identity
-      #
-      #end
+      def validate_credentials_include_azure_token
+        # check that id token field exists and has some value
+        raise Errors::Authentication::RequestBody::MissingRequestParam, JWT_FIELD_NAME unless decoded_credentials.include?(JWT_FIELD_NAME) &&
+          !decoded_credentials[JWT_FIELD_NAME].empty?
+      end
+
+      def validate_azure_token
+        decoded_token
+      end
+
+      def decoded_token
+        @decoded_token ||= @verify_and_decode_token.call(
+          provider_uri:     provider_uri,
+          token_jwt:        decoded_credentials[JWT_FIELD_NAME],
+          claims_to_verify: {
+            verify_iss: true,
+            iss:        provider_uri
+          }
+        )
+      end
+
+      def validate_required_token_fields_exist
+        validate_token_field_exists( "xms_mirid")
+        validate_token_field_exists("oid")
+      end
+
+      def validate_token_field_exists field_name
+        @logger.debug(Log::ValidatingTokenFieldExists.new(field_name))
+        raise Err::TokenFieldNotFoundOrEmpty, field_name if decoded_token[field_name].to_s.empty?
```

Comment:
I don't think it is enough to validate that the decoded token is not empty. In the test plan meeting, we also talked about validating that:
1. The token is readable and not corrupted (this might require further research on how a token can be corrupted)
2. xms_mirid is in the format we expect. We could check this by counting slashes if consistent or at least ensuring all parms in xms_mirid we expect to be there are (resource_group/subscription_id)

Reply:
1. If the token is not a valid one then we will fail in the decoding so we're in the clear on this.
2. I think it will be easier to do this in a different PR. This one is already pretty big and your suggestion requires some UTs and more logic. I prefer we merge this and get to the xms_mirid validation after that.

Reply:
okay, adding this to our running list of things to add